src/test/java/org/jabref/logic/search/query/SearchQueryLuceneConversionTest.java

```diff
@@ -10,64 +12,55 @@
+                Arguments.of("(content:on annotations:on) (+(content:two annotations:two) +(content:three annotations:three))", "one OR (two AND three)"),
 
-                // not converted, because wrong syntax for JabRef 5.x
-                // Arguments.of("( author:miller OR title:\"image processing\" OR keywords:\"image processing\" ) AND NOT author:brown AND NOT author:blue", "(author = miller or title|keywords = \"image processing\") and not author = brown and != author = blue"),
+                Arguments.of("(-content:term) (-annotations:term)", "any != term"),
+                Arguments.of("(-content:term) (-annotations:term)", "any !== term"),
+                Arguments.of("(-content:term) (-annotations:term)", "any !=! term"),
+                Arguments.of("(-content:\"two term\") (-annotations:\"two term\")", "any != \"two terms\""),
+                Arguments.of("+(-content:imag) +(-annotations:process)", "content != image AND annotations != processing"),
 
-                // String with a space
-                Arguments.of("title:image\\ processing AND author:smith", "title = \"image processing\" AND author= smith"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = image"),
+                Arguments.of("content:\"imag process\" annotations:\"imag process\"", "\"image processing\" AND author = smith"),
+                Arguments.of("+(content:imag annotations:imag) +(content:process annotations:process)", "image AND (title = term OR processing)"),
+                Arguments.of("(content:imag annotations:imag) (content:process annotations:process)", "image OR (title = term OR processing)"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = \"image processing\" AND author = smith"),
 
-                // We renamed fields
-                Arguments.of("any:somecontent", "anyfield = somecontent"),
-                Arguments.of("keywords:somekeyword", "anykeyword = somekeyword"),
-                Arguments.of("citationkey:somebibtexkey", "key = somebibtexkey")
-        );
-    }
+                Arguments.of("content:neighbou?r annotations:neighbou?r", "neighbou?r"),
+                Arguments.of("content:neighbo* annotations:neighbo*", "neighbo*"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = neighbou?r"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "(title == chocolate) OR (author == smith)"),
 
-    @ParameterizedTest
-    @MethodSource
-    void transformationNormal(String expected, String query) {
-        String result = SearchQueryConversion.searchToLucene(query);
-        assertEquals(expected, result);
-    }
-
-    public static Stream<Arguments> transformationRegularExpression() {
-        return Stream.of(
-                Arguments.of("any:* AND -groups:/.+/", "groups != .+"),
-                Arguments.of("( any:* AND -groups:/.+/ ) AND ( any:* AND -readstatus:/.+/ )", "groups != .+ and readstatus != .+"),
```

Comment:
This should also be kept: non-empty group and set read-status

Reply:
> This should also be kept: non-empty group and set read-status

This is only for searching linked files using Lucene. The searchable fields in linked files are `content` and `annotations`. Searching in bib fields is handled by SQL queries. I will add those test cases to `org.jabref.logic.search.query.SearchQuerySQLConversionTest`.